### PR TITLE
ESP supports API basepath

### PR DESCRIPTION
**Pull request by mangchiandjjoe**
This is the recovery for the PR #287, the fork repository was accidentally deleted.

### CODE REVIEW

File: src/nginx/module.cc [hunk 1]
```diff
@@ -49,6 +49,41 @@ namespace google {
...
+      lc->esp->ReWriteURL(ngx_str_to_std(r->unparsed_uri), &unparsed_uri);
+
+  if (unparsed_uri_action == ApiManager::ApiBasepathRewriteAction::REWRITE) {
+    ngx_str_copy_from_std(r->pool, unparsed_uri, &r->unparsed_uri);
+    ngx_str_copy_from_std(r->pool,
+                          ngx_str_to_std(r->method_name) + " " + unparsed_uri +
+                              " " + ngx_str_to_std(r->http_protocol),
+                          &r->request_line);
+  } else if (unparsed_uri_action ==
+             ApiManager::ApiBasepathRewriteAction::REJECT) {
```
Comment (qiwzhang) [inline src/nginx/module.cc]: should we just "return" for  None case?

**Comment by mangchiandjjoe** [comment]
PTAL

File: src/nginx/module.cc [hunk 2]
```diff
@@ -70,6 +70,8 @@ ApiManager::ApiBasepathRewriteAction ngx_esp_rewrite_uri(
   } else if (unparsed_uri_action ==
              ApiManager::ApiBasepathRewriteAction::REJECT) {
     return unparsed_uri_action;
+  } else {
```
Comment (qiwzhang) [inline src/nginx/module.cc]: how about just one "else" block for line 70-75

Comment (mangchiandjjoe) [reply src/nginx/module.cc]: I got confused. If the unparsed_uri_action is ::NONE, we need to rewrite r->uri. 
So we should not return for ::NONE

I will revert the change.

File: src/nginx/module.cc (hunk 1 above)
Comment (qiwzhang) [reply src/nginx/module.cc]: I think if action == NONE,  we should return it too.  Only REWRITE case, we need to run the rest of code.

So the code at line 64 can be:

if (uri_action != REWRITE) return uri_action;

Comment (mangchiandjjoe) [reply src/nginx/module.cc]: My bad. You are right. I updated.

**Pull request merged**